### PR TITLE
Add sectional curvature for the multinomials + missing tests

### PR DESCRIPTION
**Pull request by ninamiolane**
This PR:
- adds the closed form formula for the constant sectional curvature of the multinomial manifold
- adds new format of tests for the multinomial distributions (using LevelSetTestCase) and test_data.
- change the output of the function `sample` of the multinomial distributions:
before, it was returning the list of categorical labels across the n_draws
not it is returning the counts for each categorical labels across the n_draws.

## Checklist

- [x] My pull request has a clear and explanatory title.
- [x] If neccessary, my code is [vectorized](https://www.geeksforgeeks.org/vectorization-in-python/).
- [x] I have added apropriate unit tests.
- [ ] I have made sure the code passes all unit tests. (refer to comment below)
- [x] My PR follows [PEP8](https://peps.python.org/pep-0008/) guidelines. (refer to comment below)
- [x] My PR follows [geomstats coding style](https://github.com/geomstats/geomstats/blob/master/docs/contributing.rst#coding-style-guidelines) and API.
- [x] My code is properly documented and I made sure the documentation renders properly. ([Link](https://github.com/geomstats/geomstats/blob/master/docs/contributing.rst#documentation))
- [x] I have linked to issues and PRs that are relevant to this PR.

### CODE REVIEW

File: tests/tests_geomstats/test_multinomial.py [hunk 1]
```diff
@@ -1,153 +1,131 @@
...
+    skip_test_extrinsic_after_intrinsic = True
+    skip_test_intrinsic_after_extrinsic = True
+    testing_data = MultinomialTestData()
+
+    def test_sample_shape(self, dim, n_draws, point, n_samples, expected):
+        self.assertAllClose(
+            self.Space(dim, n_draws).sample(point, n_samples).shape, expected
+        )
 
+    def test_projection(self, dim, n_draws):
```
Comment (lpereira95) [inline tests/tests_geomstats/test_multinomial.py:25]: Probably we can abstract this test later (and also `test_to_tangent`). That is being passed as `smoke`, but in fact it is a `random` test.

**Review by lpereira95 — APPROVED** [review]
Thanks @ninamiolane. Just some little comments..

File: geomstats/information_geometry/multinomial.py [hunk 2]
```diff
@@ -379,3 +385,57 @@ def path(t):
...
+        sectional curvature given by :math:`K = 2 \sqrt{n}`.
+
+        Parameters
+        ----------
+        tangent_vec_a : array-like, shape=[..., n, n]
+            Tangent vector at `base_point`.
+        tangent_vec_b : array-like, shape=[..., n, n]
+            Tangent vector at `base_point`.
+        base_point : array-like, shape=[..., n, n]
+            Point in the manifold.
```
[the remark this answers is not in the document]
Comment (alebrigant) [reply geomstats/information_geometry/multinomial.py:412]: shapes should be `[..., dim + 1]` ?

File: geomstats/information_geometry/multinomial.py [hunk 3]
```diff
@@ -379,3 +385,57 @@ def path(t):
...
+
+    def sectional_curvature(self, tangent_vec_a, tangent_vec_b, base_point=None):
+        r"""Compute the sectional curvature.
+
+        In the literature sectional curvature is noted K.
+
+        For two orthonormal tangent vectors :math:`x,y` at a base point,
+        the sectional curvature is defined by :math:`K(x,y) = <R(x, y)x, y>`.
+        For non-orthonormal vectors, it is
+        :math:`K(x,y) = <R(x, y)x, y> / \\|x \wedge y\\|^2`.
```
[the remark this answers is not in the document]
Comment (alebrigant) [reply geomstats/information_geometry/multinomial.py:397]: Maybe this formula would be more clear ?
`:math: K(x,y) = <R(x, y)x, y> / (<x, x><y, y> - <x, y>^2)`

**Review by alebrigant — APPROVED** [review]
Looks great to me, thanks !! not much to add to Luis' comments.

File: geomstats/information_geometry/multinomial.py [hunk 4]
```diff
@@ -128,15 +129,20 @@ def sample(self, point, n_samples=1):
...
         point = gs.to_ndarray(point, to_ndim=2)
         samples = []
         for param in point:
             counts = multinomial.rvs(self.n_draws, param, size=n_samples)
-            samples.append(gs.argmax(counts, axis=-1))
+            if n_samples == 1:
+                samples.append(counts[0])
+            else:
+                samples.append(counts)
         return samples[0] if len(point) == 1 else gs.stack(samples)
```
[the remark this answers is not in the document]
Comment (ninamiolane) [reply geomstats/information_geometry/multinomial.py:146]: Done!

File: geomstats/information_geometry/multinomial.py [hunk 5]
```diff
@@ -32,7 +32,7 @@ class MultinomialDistributions(InformationManifoldMixin, LevelSet):
         Embedding manifold.
     """
 
-    def __init__(self, dim, n_draws):
+    def __init__(self, dim, n_draws, **kwargs):
         super().__init__(
```
[the remark this answers is not in the document]
Comment (ninamiolane) [reply geomstats/information_geometry/multinomial.py:36]: Done

File: geomstats/information_geometry/multinomial.py [hunk 6]
```diff
@@ -32,7 +32,7 @@ class MultinomialDistributions(InformationManifoldMixin, LevelSet):
         Embedding manifold.
     """
 
-    def __init__(self, dim, n_draws):
+    def __init__(self, dim, n_draws, **kwargs):
```
[the remark this answers is not in the document]
Comment (ninamiolane) [reply geomstats/information_geometry/multinomial.py:35]: Done

File: geomstats/information_geometry/multinomial.py [hunk 7]
```diff
@@ -128,15 +129,20 @@ def sample(self, point, n_samples=1):
 
         Returns
         -------
-        samples : array-like, shape=[..., n_samples]
+        samples : array-like, shape=[..., n_samples, dim + 1]
             Samples from multinomial distributions.
+            Note that this can be of shape [n_points, n_samples, dim + 1] if
+            several points and several samples are provided as inputs.
         """
         geomstats.errors.check_belongs(point, self)
```
[the remark this answers is not in the document]
Comment (ninamiolane) [reply geomstats/information_geometry/multinomial.py:137]: Removed!

File: geomstats/information_geometry/multinomial.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (ninamiolane) [reply geomstats/information_geometry/multinomial.py:36]: Done!

File: geomstats/information_geometry/multinomial.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (ninamiolane) [reply geomstats/information_geometry/multinomial.py:397]: Done, also corrected in RiemannianMetric. Thanks for catching this!

File: geomstats/information_geometry/multinomial.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ninamiolane) [reply geomstats/information_geometry/multinomial.py:412]: Done!

File: tests/tests_geomstats/test_multinomial.py [hunk 8]
```diff
@@ -1,153 +1,131 @@
...
+        """Test projection.
         Test that result belongs to the simplex.
```
[the remark this answers is not in the document]
Comment (ninamiolane) [reply tests/tests_geomstats/test_multinomial.py:27]: Yes!

File: tests/tests_geomstats/test_multinomial.py (hunk 1 above)
Comment (ninamiolane) [reply tests/tests_geomstats/test_multinomial.py:25]: Yes, this comes from the fact that the tests were written in the old style format. I've spent some time to adapt them a little, but more needs to be done.

**Comment by ninamiolane** [comment]
Ignoring the DeepSource error since it's about making a method static, but we want to keep the `self` for consistency across manifolds/metrics

**Pull request merged**